### PR TITLE
dist: tools: factor out changed files collection in check scripts

### PR DESCRIPTION
**Pull request by kaspar030**
There was a lot of hacky, duplicated shell code. This PR removes the duplication by sourcing common a common file.

### CODE REVIEW

File: dist/tools/ci/changed_files.sh [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# Copyright 2017 Kaspar Schleiser <kaspar@schleiser.de>
+# Copyright 2014 Ludwig Knüpfer <ludwig.knuepfer@fu-berlin.de>
+# Copyright 2014 Hinnerk van Bruinehsen <h.v.bruinehsen@fu-berlin.de>
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+
+changed_files() {
+    : ${FILEREGEX:='\.([CcHh]|[ch]pp)$'}
```
Comment (miri64) [inline dist/tools/ci/changed_files.sh]: what does the leading `:` do?

File: dist/tools/cppcheck/check.sh [hunk 2]
```diff
@@ -1,12 +1,15 @@
...
 
+# Copyright 2017 Kaspar Schleiser <kaspar@schleiser.de>
 # Copyright 2014 Ludwig Knüpfer <ludwig.knuepfer@fu-berlin.de>
 # Copyright 2014 Hinnerk van Bruinehsen <h.v.bruinehsen@fu-berlin.de>
 #
 # This file is subject to the terms and conditions of the GNU Lesser
 # General Public License v2.1. See the file LICENSE in the top level
 # directory for more details.
 
+. ${RIOTBASE:+${RIOTBASE}/}dist/tools/ci/changed_files.sh
```
Comment (miri64) [inline dist/tools/cppcheck/check.sh]: I prefer `source <path>` for more clarity for non-bash magicians.

File: dist/tools/cppcheck/check.sh [hunk 3]
```diff
@@ -1,12 +1,15 @@
 #!/bin/sh
 
+# Copyright 2017 Kaspar Schleiser <kaspar@schleiser.de>
```
Comment (miri64) [inline dist/tools/cppcheck/check.sh]: Copyright note on code removal ^^?

File: dist/tools/ci/changed_files.sh (hunk 1 above)
Comment (gebart) [inline dist/tools/ci/changed_files.sh]: it's like a NOP in bash. It is used when you want the side effects of, for example, `${FILEREGEX:='\.([CcHh]|[ch]pp)$'}`, which is to set `FILEREGEX = '\.([CcHh]|[ch]pp)$'` if it was not already set before.

Comment (kaspar030) [inline dist/tools/ci/changed_files.sh]: > it's like a NOP in bash.

Nice thing about it is that it is specified by POSIX and thus works mostly everywhere.
It is the nicest (portable) way of doing ```FOO?=bar```. The next best thing is ```FOO=${FOO:-bar}```, which IMO gets ugly as soon as ```bar``` gets more complicated.

File: dist/tools/cppcheck/check.sh (hunk 2 above)
Comment (kaspar030) [inline dist/tools/cppcheck/check.sh]: uff. We need to come up with "style commenting guidelines". Those subjective "I prefer style X over Y" IMO need to get some substance.

Your argument could also be used against usage of C's ternary operator.

Comment (kaspar030) [inline dist/tools/cppcheck/check.sh]: Anyhow, I don't care, so I changed it.

File: dist/tools/cppcheck/check.sh (hunk 3 above)
Comment (kaspar030) [inline dist/tools/cppcheck/check.sh]: Well, I didn't remove Ludwig and Hinnerk, now that their code is gone. ;)

File: dist/tools/cppcheck/check.sh (hunk 2 above)
Comment (kaspar030) [inline dist/tools/cppcheck/check.sh]: hmpf, seems like ```source``` is actually unusable for non-bash magicians.

Comment (kaspar030) [inline dist/tools/cppcheck/check.sh]: So I removed the commit changing this to "source" again.

**Comment by kaspar030** [comment]
- rebased
- comments addressed (or at least commented.)

This is blocking coccinelle support...

File: dist/tools/cppcheck/check.sh (hunk 2 above)
Comment (miri64) [inline dist/tools/cppcheck/check.sh]: Okay.

**Pull request merged**